cmd/syncthing/main.go

```diff
@@ -746,12 +745,30 @@ func syncthingMain(runtimeOptions RuntimeOptions) {
 		}
 	}
 
-	if cfg.RawCopy().OriginalVersion == 15 {
-		// The config version 15->16 migration is about handling ignores and
-		// delta indexes and requires that we drop existing indexes that
-		// have been incorrectly ignore filtered.
+	// Grab the previously running version string from the database.
+
+	miscDB := db.NewNamespacedKV(ldb, string(db.KeyTypeMiscData))
+	prevVersion, _ := miscDB.String("prevVersion")
+
+	// Strip away prerelease/beta stuff and just compare the release
+	// numbers. 0.14.44 to 0.14.45-banana is an upgrade, 0.14.45-banana to
+	// 0.14.45-pineapple is not.
+
+	prevParts := strings.Split(prevVersion, "-")
+	curParts := strings.Split(Version, "-")
+	if prevParts[0] != curParts[0] {
```

Comment:
Does that work? The first time running the miscDB does not contain `prevVersion` and thus `prevParts` should be `nil`. Also maybe use the code from the upgrade package here.

Reply:
Yeah, it works. Split always returns at least one string, which may be empty, so the first run will be considered an upgrade which is desired. The stuff in lib/upgrade is overkill, more cumbersome to when we just need the string comparison, and unexported.